### DISCUSSION TITLE
Change configuration keywords `static`,  `dynamic`, static `item` and `set`

### DISCUSSION THREAD

**Discussion by martin-rueegg** [moudey/Shell#202] [Ideas]
I've been working extensively with the configuration files and the documentation. I have found some terms to be confusing or "characterless" and would suggest to change them in order to improve understanding and readability, also avoiding lingual ambiguity or ambiguity with other terms.

The following therms I have found to be confusing.

### `set`

| Current | New | Comments |
|--|--|--|
|`set`|`settings`|The term "set" reminds me more of the actual linguistic [noun _set_](https://www.wordnik.com/words/set), rather than a short form of "settings" or "defaults" (as it was termed before). Hence, I consider `settings` to be more explanatory.

### `static` and `dynamic`

These terms are not very self-explanatory. Even after understanding the concept, I find the terms still meaningless or rather a bit confusion. `static` seems to refer to the "existing" items, as in "already built", hence "static". Whereas `dynamic` might refer to the action of "creating/building". However, the `static` section is about _changing_ items (hence also _dynamic_) while the `dynamic` section is not necessarily as _dynamic_ as it might seem.

(I'm aware that the author is not a native English speaker [neither am I, btw.] it it might be the result of automatic translation.)

My suggestion would be to deprecate the two terms and replace them as follows:

| Current | New | Comments |
|--|--|--|
| `static` | `update` | As explained above, the term `static` seems to refer to the existing _items_. However, the configuration here is really about the _change_ done to those items - in terms of appearance or function. The term "_change_", however, seems not to be a fantastic keyword for a section. Hence `update`, which is already known in the IT world, particularly from SQL, where also _existing_ entries are _changed/updated_.
| `dynamic` | `insert` | Again, the term _dynamic_ is not very explanatory. Analogous to the above entry, I'd suggest `insert` as it is adding/inserting new menuitems.

### Static `item`
| Current | New | Comments |
|--|--|--|
| `item` | ~`process`~[^1] or `change`[^2] | The main issue here is the ambiguity with the [dynamic `item`](https://nilesoft.org/docs/syntax/dynamic#item) entry. While they are similar, they are not the same. A dynamic `item` is _always_ a [item](https://nilesoft.org/docs/syntax/dynamic#item), never a [menu](https://nilesoft.org/docs/syntax/dynamic#menu). Also, it does not actually represent a [menuitem](https://nilesoft.org/docs/syntax/dynamic#menuitem), but rather the instruction on how to identify and _change_ them.[^3]

[^1]: `process` might not be a great choice. It came up due to the term [process instructions](https://github.com/metaworx/nilesoft.shell/blob/e58e68941feab873af75f55de24e232c835afef0/docs/syntax/static.html#L95) I've used in the documentation to describe, what those items do.
[^2]: `change` might be the better choice: In the `update` sections are the instructions how to `change` existing menuitems ...
[^3]: Of course, also `update` would work here, but if `update` would be used as the section name, ambiguity there must be avoided.

**Comment by moudey** [comment]
I amended by adding the following while keeping the current names

```
set | settings

static | update
{
    item | change
}

dynamic | new | insert
```

I also plan to change the name of Shell after many users suggested it.

**Reply by martin-rueegg** [reply]
> I amended by adding the following while keeping the current names

That sounds awesome!

Would you agree on the new names be the default, and the old names being "deprecated"? Then I would go and update the documentation accordingly.

> I also plan to change the name of Shell after many users suggested it.

Yes, I can understand that. Firstly, most people don't know that explorer.exe in Windows is primarily the shell, and only kinda secondary a File Explorer. So for them the name "Shell" might not make much sense. And then, **Shell** is actually not a "shell" in the technical sense, but rather a tool to modify the real shell, explorer.exe.

Do you already have a name in mind?

Is there maybe a beautiful word in Arabic for _customize_, _re-think_, _improve_, _enhance_, _make it your own_, that could be transliterated?

**Reply by moudey** [reply]
>Would you agree on the new names be the default, and the old names being "deprecated"? Then I would go and update the documentation accordingly.

But it is supposed to be published with the new version, right?

>Do you already have a name in mind?

I already have the name "dija", which is short for my little girl, Khadija.

**Reply by martin-rueegg** [reply]
> > Would you agree on the new names be the default, and the old names being "deprecated"? Then I would go and update the documentation accordingly.
> 
> But it is supposed to be published with the new version, right?

Absolutely. Do you have an estimate on when to release? There might be some other "naming" issues in the functions, that we could discuss.

**Reply by martin-rueegg** [reply]
> 
> > Do you already have a name in mind?
> 
> I already have the name "dija", which is short for my little girl, Khadija.

Sounds nice! - But you should ask her! Princesses do not like to be dubbed. And then it's confusing within your family whether you're referring to your work, or your love.... Just my two cents. :-)
